source/code/include/scxcorelib/scxsingleton.h

```diff
@@ -58,21 +58,39 @@ namespace SCXCoreLib
             If the instance is not yet created, it first creates it.
 
         */
-        static T& Instance();
+        static T& Instance()
+        {
+            if (0 == s_lockHandle.GetData())
+            {
+                throw SCXInternalErrorException(L"Tried to get a singleton instance before static initialization was completed.", SCXSRCLOCATION);
+            }
+
+            SCXThreadLock lock(*s_lockHandle, true);
+
+            if (0 == s_instance.GetData())
+            {
+                s_instance = new T();
+            }
+
+            return *s_instance;
```

Comment:
This won't work. Doing it this way creates a scenario where, depending on construction, you may have race conditions. Thus, we created new mechanisms on construction of singletons to make this "correct".